### ISSUE TITLE
SFM Gateway Roadmap vs Envoy

### ISSUE THREAD

**Issue by SamirFarhat** [Azure/service-fabric-mesh-preview#323]
Hi,
We now know that Envoy will not be provided as a side car Service, but through SFM Gateway.
I like the 'managed' idea since this bring many benefits.
The challenge now is to know how the gateway will integrate Envoy features.
Can the PG share the roadmap of features that SFMG will provide compared to all Enviy features.
Thanks

**Comment by SamirFarhat** [comment]
@ChackDan Thanks.
This would be possible when the Gateway will be optional. If the Gateway is anyway deployed, then i would avoid duplicating the service, in addition that inprefer a managed service, then my own to avoid management complexity. What bring us to SFM is that we want only to focus on our business, never more infrastructure.

**Comment by guibirow** [comment]
AFAIK, envoy is being used as an Edge Proxy only, not a real implementation of a Mesh Network, the term MESH in the product name gives us this misconception.

Using envoy as a sidecar should be a developers choice, I am glad they didn't force down on our throat this approach, would be too intrusive to our architecture design.

In case we do want to follow the MESH approach, is not clear how our envoy sidecars would be configured, it will need a control plane and a data plane for configuration and service discovery, I am not sure we will have access this data somewhere else other than a DNS service.

**Comment by SamirFarhat** [comment]
Hi again,
Envoy or Linkerd are part of the Data Plane in a Service Mesh, so what about the Control plane.
We real need a clear roadmap about the possible features of the SFM Gateway, in addition to the other features that we need to see for authentication, routing, load balancing ...

**Comment by tomkerkhove** [comment]
Agreed, maybe also some more docs on the gateway itself and how/if it's using Envoy under the hood like mentioned at Build/Ignite
